src/edu/csus/ecs/pc2/ui/ShadowCompareRunsPane.java

```diff
@@ -327,11 +343,132 @@ private ShadowCompareRunsSummaryPane getSummaryPanel() {
+                                log.log(Log.WARNING, "dynamicallyRefreshPanel received duration more than 60. It received: "+ textField.getText());
+                                throw new IllegalArgumentException("Duration value cannot be more than 60, time is: " + duration);
+                            }
+                        } catch (NumberFormatException a) {
+                            log.log(Log.WARNING, "dynamicallyRefreshPanel did not receive an integer for duration. It received: "+ textField.getText());
+                            throw new NumberFormatException("dynamicallyRefreshPanel did not receive an integer for duration. It received: " + textField.getText());
+                        }
+                        SwingUtilities.invokeLater(new Runnable() {
+                            public void run() {
+                                refreshResultsTable();
+                            }
+                        });
+                        timer = new Timer(duration * 1000, new ActionListener() {
+                            @Override
+                            public void actionPerformed(ActionEvent e) {
+                                
+
+                                SwingUtilities.invokeLater(new Runnable() {
+                                    public void run() {
+                                        refreshResultsTable();
+                                    }
+                                });
+                            }
+                        });
+                        textField.setEnabled(false);
+                        log.log(Log.INFO, "Shadow table will be dynamically refreshed every " + duration + " seconds.");  
+                        timer.start();
+                    } 
+                    else {
+                        textField.setEnabled(true);
+                        log.log(Log.INFO, " Dynamically refreshing has been stopped.");
+                        timer.stop();
+                    }
+                }
+            });
+            
+            JLabel label = new JLabel("seconds");
+            dynamicallyRefreshPanel.add(label);
+            
+            Component horizontalStrut2 = Box.createHorizontalStrut(40);
+            dynamicallyRefreshPanel.add(horizontalStrut2);
+            
+            JCheckBox missMatchCheckBox = new JCheckBox("Only Missmatched");
+            missMatchCheckBox.setToolTipText("When toggled, table will only display when pc2 and remote ccs' judgements do not match");
+            
+            missMatchCheckBox.addItemListener(new ItemListener() {
+                @Override
+                public void itemStateChanged(ItemEvent e) {
+                    if (e.getStateChange() == ItemEvent.SELECTED) {
+                        shadowController.setFilter(FILTERS.ONLY_MISSMATCH);
+                    }
+                    else {
+                        shadowController.setFilter(FILTERS.NONE);
+                    }
+                }
```

Comment:
Perhaps perform a "refresh" here if the box is checked/unchecked, so it updates right away.  I clicked it, and nothing happened until I manually pressed "Refresh".

Reply:
As you have said, users expect update immediately after checking/unchecking mismatch box. However putting refreshing there could be unexpected for the users as well as it doesn't explicitly state that it will refresh. I think a better way to do is to either maintain two tables that get swapped or to create a filtered version every time only mismatched is toggled. First suggestion could be problematic because if a judgement get resolved, I don't know how it will appear on the other table.